editor/src/main/java/net/rpgtoolkit/editor/ui/MainWindow.java

```diff
@@ -663,19 +673,50 @@ public Player openCharacter(File file) {
     return null;
   }
 
-  /**
-   * Creates a TileSet editor window for modifying the specified TileSet.
-   *
-   * @param file
-   */
-  public void openTile(File file) {
-    TileEditor testTileEditor = new TileEditor(file);
-    desktopPane.add(testTileEditor);
+  public void createNewTileset() {
+    NewTilesetDialog dialog = new NewTilesetDialog();
+    dialog.setLocationRelativeTo(this);
+    dialog.setVisible(true);
+
+    if (dialog.getValue() != null) {
+      int tileWidth = dialog.getValue()[0];
+      int tileHeight = dialog.getValue()[1];
+
+      fileChooser.resetChoosableFileFilters();
+      FileNameExtensionFilter filter = new FileNameExtensionFilter(
+              "Image Files", getImageExtensions());
+      fileChooser.setFileFilter(filter);
+
+      if (fileChooser.showOpenDialog(this) == JFileChooser.APPROVE_OPTION) {
+        File file = fileChooser.getSelectedFile();
+
+        try (FileInputStream fis = new FileInputStream(file)) {
+          BufferedImage source = ImageIO.read(fis);
+
+          TileSet tileSet = TileSetRipper.rip(source, tileWidth, tileHeight);
+
+          File tileSetFile = MainWindow.getInstance().saveByType(TileSet.class);
+          tileSet.setDescriptor(new AssetDescriptor(tileSetFile.toURI()));
+          tileSet.setName(tileSetFile.getName());
+          
+          AssetManager.getInstance().serialize(
+                  AssetManager.getInstance().getHandle(tileSet));
+          
+          openTileset(tileSetFile);
+        } catch (IOException | AssetException ex) {
+          Logger.getLogger(MainWindow.class.getName()).log(Level.SEVERE, null, ex);
+        }
+      }
+    }
   }
 
   public void openTileset(File file) {
-    TileSet tileSet = new TileSet(file);
+    TileSetCache tileSetCache = TileSetCache.getInstance();

```

Comment:
Will want to update this now that TileSetCache is static. (Will actually be simpler!)